cms/models.py

```diff
@@ -466,3 +514,172 @@ def get_context(self, request, *args, **kwargs):
             **super().get_context(request, *args, **kwargs),
             "site_name": settings.SITE_NAME,
         }
+
+
+class InnerResourcePage(ResourcePage):
+    """
+    Interior page - same general format as ResourcePage, but can be nested under a ResourcePage
+    """
+
+    template = "resource_page.html"
+    parent_page_types = ["ResourcePage"]
+
+
+class FlexiblePricingRequestForm(AbstractForm):
+    intro = RichTextField(blank=True)
+    thank_you_text = RichTextField(blank=True)
+    guest_text = RichTextField(
+        null=True,
+        blank=True,
+        help_text="What to show if the user isn't logged in.",
+    )
+    application_processing_text = RichTextField(
+        null=True,
+        blank=True,
+        help_text="What to show if the user's request is being processed.",
+    )
+    application_approved_text = RichTextField(
+        null=True,
+        blank=True,
+        help_text="What to show if the user's request has been approved.",
+    )
+    application_denied_text = RichTextField(
+        null=True,
+        blank=True,
+        help_text="What to show if the user's request has been denied.",
+    )
+
+    content_panels = AbstractForm.content_panels + [
+        FieldPanel("intro"),
+        FieldPanel("guest_text"),
+        InlinePanel("form_fields", label="Form Fields"),
+        FieldPanel("thank_you_text"),
+        FieldPanel("application_processing_text"),
+        FieldPanel("application_approved_text"),
+        FieldPanel("application_denied_text"),
+    ]
+
+    parent_page_types = ["cms.HomePage", "cms.ResourcePage", "cms.CoursePage"]
+
+    form_builder = FlexiblePricingFormBuilder
+    template_name = "flexiblepricing/flexible_pricing_request_form.html"
+    landing_page_template = "flexiblepricing/flexible_pricing_request_form_landing.html"
+
+    def get_submission_class(self):
+        return FlexiblePricingRequestSubmission
+
+    def get_template(self, request, *args, **kwargs):
```

Comment:
I think the template name is already all set from this line `template_name = "flexiblepricing/flexible_pricing_request_form.html"`.  It seems that the method is only useful if you want to change the template dynamically.